src/index.ts

```diff
@@ -1,28 +1,40 @@
-import * as Crypto from "./crypto.js";
-import * as Utils from "./utils.js";
+import Account, { Keychain } from "./account.js";
 import * as Api from "./api.js";
-import { DirectEndpoint, Endpoint, WalletRPCEndpoint } from "./endpoint.js";
-import { ArchethicRPCClient } from "./api/wallet_rpc.js";
 import { NodeRPCClient } from "./api/node_rpc.js";
+import { ConnectionState } from "./api/types.js";
+import { AWCWebBrowserExtensionStreamChannel } from "./api/wallet_rpc.browserextension.js";
+import { ArchethicWalletClient } from "./api/wallet_rpc.js";
+import * as Crypto from "./crypto.js";
+import { AWCEndpoint, Endpoint } from "./endpoint.js";
 import Network from "./network.js";
 import Transaction from "./transaction.js";
-import Account, { Keychain } from "./account.js";
+import * as Utils from "./utils.js";
 
-export { Utils, Crypto, Keychain };
+export {
+  AWCWebBrowserExtensionStreamChannel,
+  ArchethicWalletClient,
+  ConnectionState,
+  Crypto,
+  Keychain,
+  Utils
+};
 
 export default class Archethic {
-  endpoint: DirectEndpoint | WalletRPCEndpoint;
-  rpcWallet: ArchethicRPCClient | undefined;
+  endpoint: Endpoint;
+  rpcWallet: ArchethicWalletClient | undefined;
   rpcNode: NodeRPCClient | undefined;
   transaction: Transaction;
   nearestEndpoints: Set<string>;
   account: Account;
   network: Network;
 
-  constructor(endpoint: string) {
+  /**
+   * @param endpoint if undefined, endpoint will be resolved using ArchethicWalletClient.
+   */
+  constructor(endpoint: string | undefined) {
```

Comment:
Probably the same remark/question as above

Reply:
I Agree with you.

We could directly pass an `Endpoint` instance here. This will be mandatory to extract `AWCWebBrowserExtention` from here.

Reply:
After more thought, this would be a breaking change. 

Looking at the number of PRs waiting for validation, I think that change should wait a bit.